nautobot/core/utils/lookup.py

```diff
@@ -276,11 +276,16 @@ def get_table_class_string_from_view_name(view_name):
     Returns:
         table_class_name (String): The name of the model table class or None e.g. LocationTable, CircuitTable
     """
-    model = get_model_for_view_name(view_name)
-    if model:
-        table_class = get_table_for_model(model)
-        if table_class:
-            return table_class.__name__
+    app_label, model_name = view_name.split(":")
+    model_name = model_name.split("_")[0]
+    model_name = f"{app_label}.{model_name}"
+    # model name should be in the form of "dcim.location".
+    model = get_model_from_name(model_name)
+    view_class = get_view_for_model(model, "List")
```

Comment:
This feels overly convoluted to me. Given we have `view_name`, can't we resolve that directly to the `view_class` without needing to go by way of the `model`?